sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -120,14 +120,14 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
         return -EINVAL;
     }
 #if IS_USED(MODULE_ZTIMER_USEC)
-    ztimer_t timeout_timer;
+    ztimer_t timeout_timer = { .base = { .next = NULL } };
```

Comment:
If the initialsation is done with ` = {.callback = _callback_put}` the rest of the struct should be initialised (to zero) as well.

http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf (p.:142  pdf.:160  l:28 there is an example how to handel partial initialisation)